shared_model/interfaces/commands/add_peer.hpp

```diff
@@ -45,16 +45,15 @@ namespace shared_model {
       std::string toString() const override {
         return detail::PrettyStringBuilder()
             .init("AddPeer")
-            .append("peer_key", peerKey().toString())
+            .append(peerKey().toString())
             .append("peer_address", peerAddress())
             .finalize();
       }
 
       OldModelType *makeOldModel() const override {
         auto oldModel = new iroha::model::AddPeer;
         oldModel->address = peerAddress();
-        // TODO: check if this is right
-        oldModel->peer_key.from_string(peerKey().blob());
+        oldModel->peer_key.from_string(peerKey().makeOldModel()->blob());
```

Comment:
Isn't here a memory leak?

Reply:
Any idea how to make it ?

Reply:
I guess only via explicit assignment, e.g:
```cpp
auto old = std::make_unique<..>(peerKey().makeOldModel());
.. old->blob()
```

Reply:
There is a memory leak. To remove all memory leaks just do not use raw new, but use smart pointers.